testGameSearch/GameViewModel.swift

```diff
@@ -12,7 +12,7 @@ import Combine
 class GameViewModel: ObservableObject {
     
     private var gameService: GameService
-    @Published var game: Game?
+    @Published var games: [Game] = .init()
```

Comment:
Arrays can be empty, so we don't need to use an optional this time. We can just initialise an empty array

Reply:
Gonna read up on when I need to init. Still gives me problems because I never know what needs it

Reply:
All non-optional properties must have a value before the parent object's initialiser can return, whether that's by passing in values at the call-site in an init function or by hard-coding default values in the class/struct itself.

In the original example you can't know what Game object you will have when the VM is initialised, so you can't pass in the value when you create the VM. This means your choices are to leave it nil and make it optional, or hard code init a blank Game struct. The problem with the second approach is that a Game needs a bunch of information for it's own initialiser. Normally the JSONDecoder passes this information in from the API for us so to do it ourselves we would need to call `Game(id: Int, name: String, summary: String?, platforms: [Game.Platforms])` and provide all that info. This is a pain in the butt. You would pick the optional in this case.

Since you're looking for an array this is a bit easier. You might not know which Games you will have when the VM loads, but the initialiser is not looking for a Game, it's looking for an Array. You can always init an empty array so long as you tell the compiler which type the Elements should be. so even though Game() won't work, [Game]() will!

Hopefully this was at least a little bit helpful!